src/main/java/com/github/lordrex34/config/util/ConfigPropertyRegistry.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2017 Reginald Ravenhorst <lordrex34@gmail.com>
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.github.lordrex34.config.util;
+
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A reload supporting registry used to check duplicated properties.
+ * @author lord_rex
+ */
+public final class ConfigPropertyRegistry
+{
+	private static final Logger LOGGER = LoggerFactory.getLogger(ConfigPropertyRegistry.class);
+	
+	/** Properties registry, that is used for misplaced configuration indication. */
+	private static final Map<String, Map<Path, Set<String>>> PROPERTIES = new TreeMap<>();
```

Comment:
(btw this way this can't be tested properly... if this would have been an instance, it could have been created newly for a new test... now unless you always empty everything (which is a lot of work for nothing essentially) you will have random leftover from previous runs...)

Reply:
Done.